sandbox/anvi-script-filter-hmm-hit-table

```diff
@@ -0,0 +1,223 @@
+#!/usr/bin/env python
+# -*- coding: utf-8
+
+import sys
+import os
+import numpy as np
+import pandas as pd
+
+import anvio
+import anvio.data.hmm
+import anvio.db as db
+import anvio.hmmops as hmmops
+import anvio.bamops as bamops
+import anvio.terminal as terminal
+import anvio.filesnpaths as filesnpaths
+
+from anvio.dbops import ContigsDatabase
+from anvio.parsers import parser_modules
+from anvio.argparse import ArgumentParser
+from anvio.tables.hmmhits import TablesForHMMHits
+from anvio.errors import ConfigError, FilesNPathsError
+
+__author__ = "Developers of anvi'o (see AUTHORS.txt)"
+__copyright__ = "Copyleft 2015-2020, the Meren Lab (http://merenlab.org/)"
+__credits__ = []
+__license__ = "GPL 3.0"
+__version__ = anvio.__version__
+__maintainer__ = "Matthew S. Schechter"
+__email__ = "mschechter@uchicago.edu"
+__provides__ = ["contigs-db", "hmm-hits"]
+__requires__ = ["contigs-db","hmm-source", "hmm-hits"]
+__description__ = ("Add or remove entries in a contigDB hmm_hits table.")
+
+pp = terminal.pretty_print
+
+class FilterHmmHitsTable(object):
+    """A class to filter hmm_hits from domtblout from hmmsearch."""
+    def __init__(self, args, quiet=False, run=terminal.Run(), progress=terminal.Progress()):
+        self.args = args
+        self.run = run
+        self.progress = progress
+
+        A = lambda x: args.__dict__[x] if x in args.__dict__ else None
+        self.contigs_db_path=A("contigs_db")
+        self.domtblout=A("domtblout")
+        self.hmm_source=A("hmm_source")
+        self.target_coverage=A("target_coverage")
+        self.query_coverage=A("query_coverage")
+        self.list_hmm_sources=A("list_hmm_sources")
+
+        if self.list_hmm_sources:
+            ContigsDatabase(self.contigs_db_path).list_available_hmm_sources()
+            sys.exit()
+
+        self.sanity_checks()
+
+        self.import_domtblout()
+
+        x = self.filter_domtblout()
+
+        self.get_hmm_hit_table(hmmsearch_tbl=x)
+
+        # remove the tmp file
+        os.remove(x)
```

Comment:
One thing we usually like to do with classes is to keep the 'work' that the class is doing separate from the initialization code. That way, you can get an instance of the class without always having to go through all those other steps, if you need to. It makes the class (even) more modular and generalizable. 

In this case, I would take all these functions out of `__init__` and put them into a separate, "driver" function that can be called by the `main` code below after a FilterHmmHitsTable() instance is initialized. 

I hope that made sense :/ It would probably be better to see an example? Then you can decide if/how to change your code to follow this pattern. 

Here is an example, in `kegg.py` you can see that the `RunKOfams` class's `__init__` only grabs arguments and does sanity checks. 
![image](https://user-images.githubusercontent.com/10360586/108148982-89405780-7097-11eb-9e59-cbfd13d0cfb2.png)

Then later in the code there is a driver function `process_kofam_hmms()` that actually does all the work. So in `anvi-run-kegg-kofams`, the script 1) initializes and 2) calls that driver:
```
def main(args):
    p = kegg.RunKOfams(args)
    p.process_kofam_hmms()
```

Reply:
Addressed [here](https://github.com/merenlab/anvio/pull/1678/commits/504d0ca3d7cf99147523da27c70289e85705699c)